src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/HeaderFileBuilder.java

```diff
@@ -122,61 +130,115 @@ private void emitFunctionWrapper(Constant mhConstant, String javaName, String na
         incrAlign();
         indent();
         append(MEMBER_MODS + " ");
-        append(functionInfo.methodType().returnType().getSimpleName() + " " + javaName + " (");
+        MethodType declType = functionInfo.methodType();
+        List<String> paramNames = functionInfo.parameterNames().get();
+        if (functionInfo.methodType().returnType().equals(MemorySegment.class)) {
+            // needs allocator parameter
+            declType = declType.insertParameterTypes(0, SegmentAllocator.class);
+            paramNames = new ArrayList<>(paramNames);
+            paramNames.add("allocator");
```

Comment:
Shouldn't this be added as the first element rather than the last?

Reply:
uhm, yes